RegisterPayments/handler.ts

```diff
@@ -0,0 +1,358 @@
+import { Context } from "@azure/functions";
+import * as express from "express";
+import { Response } from "express";
+import { sequenceS } from "fp-ts/lib/Apply";
+import { Either, either, fromOption, right } from "fp-ts/lib/Either";
+import { fromNullable } from "fp-ts/lib/Option";
+import { ContextMiddleware } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import { RequiredBodyPayloadMiddleware } from "io-functions-commons/dist/src/utils/middlewares/required_body_payload";
+import {
+  withRequestMiddlewares,
+  wrapRequestHandler
+} from "io-functions-commons/dist/src/utils/request_middleware";
+import {
+  IResponse,
+  IResponseErrorValidation,
+  IResponseSuccessJson,
+  ResponseSuccessJson
+} from "italia-ts-commons/lib/responses";
+import { DOMParser } from "xmldom";
+import { ResultEnum as ErrorResultEnum } from "../generated/definitions/ErrorResponse";
+import { RegisterPaymentNotify } from "../generated/definitions/RegisterPaymentNotify";
+import {
+  ResultEnum as SuccessResultEnum,
+  SuccessResponse
+} from "../generated/definitions/SuccessResponse";
+import { IBlobStorageService } from "../utils/blobStorage";
+import {
+  BasicAuthMiddleware,
+  IAzureBasicAuth,
+  IBasicAuthParams
+} from "../utils/middlewares/authorization";
+
+export interface IResponsePaymentError
+  extends IResponse<"IResponsePaymentError"> {}
+const ResponsePaymentError = (
+  title: string,
+  detail: string
+): IResponsePaymentError => ({
+  apply: (res: Response) =>
+    res
+      .status(500)
+      .set("Content-Type", "application/problem+json")
+      .json({ result: ErrorResultEnum.KO }),
+  detail: `${title}: ${detail}`,
+  kind: "IResponsePaymentError"
+});
+
+type IRegisterPaymentHandler = (
+  context: Context,
+  auth: IAzureBasicAuth,
+  registerPaymentNotify: RegisterPaymentNotify
+) => Promise<
+  // tslint:disable-next-line: max-union-size
+  | IResponseSuccessJson<SuccessResponse>
+  | IResponsePaymentError
+  | IResponseErrorValidation
+>;
+
+function getElementTextContent(e: Element): string | undefined {
+  return e.textContent?.trim();
+}
+
+const RT_NAMESPACE = "http://www.digitpa.gov.it/schemas/2011/Pagamenti/";
+
+function getFieldFromElement(
+  element: Element,
+  fieldName: string
+): Either<Error, string> {
+  return fromOption(new Error(`Missing field '${fieldName}'`))(
+    fromNullable(
+      element.getElementsByTagNameNS(RT_NAMESPACE, fieldName).item(0)
+    )
+      .orElse(() =>
+        // Negative RT haven't namespace on elements
+        fromNullable(element.getElementsByTagName(fieldName).item(0))
+      )
+      .mapNullable(getElementTextContent)
+  );
+}
+
+function parseDatiPagamento(
+  xmlDocument: Document
+): Either<
+  Error,
+  {
```

Comment:
can you extract this type into its own named type / interface ?

Reply:
Fixed in 05b02f7